packages/delivery-xml-http-request/delivery.js

```diff
@@ -11,14 +20,17 @@ module.exports = (client, win = window) => ({
         if (req.readyState === win.XMLHttpRequest.DONE) {
           const status = req.status
           if (status === 0 || status >= 400) {
-            client._logger.error('Event failed to send…')
-            if (body.length > 10e5) {
-              client._logger.warn(`Event oversized (${(body.length / 10e5).toFixed(2)} MB)`)
-            }
+            deliveryFailed(client, body, cb)
+          } else {
+            cb(null)
           }
-          cb(null)
         }
       }
+
+      req.onerror = function () {
+        deliveryFailed(client, body, cb)
+      }
```

Comment:
I don't think we actually need this and it would cause the callback to be called twice (once here & once in the `readystatechange` listener)

The very first thing that happens when a request errors is for the `readyState` to be set to `done`:

> The request error steps for an [XMLHttpRequest](https://xhr.spec.whatwg.org/#xmlhttprequest) object xhr, event, and optionally exception are:
> 1. Set xhr’s [state](https://xhr.spec.whatwg.org/#concept-xmlhttprequest-state) to done.
> &hellip;
>
> https://xhr.spec.whatwg.org/#request-error-steps

Testing this locally, I couldn't find a scenario where `onerror` fires but `readystatechange` doesn't

`onerror` would be good if it provided access to the exception, but unfortunately that's only possible when it's a synchronous request

So I think we can safely remove this

```suggestion
```